src/alignment/poa.rs

```diff
@@ -766,54 +615,207 @@ impl<F: MatchFunc> Poa<F> {
                     break;
                 }
                 let max_cell = if prevs.is_empty() {
-                    TracebackCell {
-                        score: traceback.get(0, j - 1).score
-                            + self.scoring.match_fn.score(r, *query_base),
-                        op: AlignmentOperation::Match(None),
-                    }
+                    traceback.get(0, j - 1) + self.scoring.match_fn.score(r, *query_base)
                 } else {
-                    let mut max_cell = TracebackCell {
-                        score: MIN_SCORE,
-                        op: AlignmentOperation::Match(None),
-                    };
+                    let mut max_cell = MIN_SCORE;
                     for prev_node in &prevs {
                         let i_p: usize = prev_node.index() + 1; // index of previous node
                         max_cell = max(
                             max_cell,
                             max(
-                                TracebackCell {
-                                    score: traceback.get(i_p, j - 1).score
-                                        + self.scoring.match_fn.score(r, *query_base),
-                                    op: AlignmentOperation::Match(Some((i_p - 1, i - 1))),
-                                },
-                                TracebackCell {
-                                    score: traceback.get(i_p, j).score + self.scoring.gap_open,
-                                    op: AlignmentOperation::Del(Some((i_p - 1, i))),
-                                },
+                                traceback.get(i_p, j - 1)
+                                    + self.scoring.match_fn.score(r, *query_base),
+                                traceback.get(i_p, j) + self.scoring.gap_open,
                             ),
                         );
                     }
                     max_cell
                 };
 
-                let score = max(
-                    max_cell,
-                    TracebackCell {
-                        score: traceback.get(i, j - 1).score + self.scoring.gap_open,
-                        op: AlignmentOperation::Ins(Some(i - 1)),
-                    },
-                );
-                if score.score > max_score_for_row {
+                let score = max(max_cell, traceback.get(i, j - 1) + self.scoring.gap_open);
+                if score > max_score_for_row {
                     max_scoring_j = j;
-                    max_score_for_row = score.score;
+                    max_score_for_row = score;
                 }
                 traceback.set(i, j, score);
             }
         }
-
         traceback
     }
-
+    // Recalculate the alignment using the traceback
+    // We need this in poa because we have to use the scoring
+    pub fn recalculate_alignment(&self, traceback: &Traceback) -> Alignment {
```

Comment:
Does this method need to be public? Or shall it be used just internally? In the latter case, please remove the pub, in the former case, please add a proper doc comment including a runtime complexity statement (O(...)).

Reply:
I suppose none of the functions in the Poa struct need to be public since the aligner uses the struct internally. I made the new function public just to keep it consistent with the others.
I added a doc comment, including the runtime complexity. Let me know if it looks okay.
Thank you.

Reply:
even if other functions should be private, maybe going forward private is the way to go? anyway, I really like this PR, as I run the POA function many times in a loop and this should speed up my performance, thanks for the changes :) 

one question I have: the test output seems to have changed in `test_yclip_suffix_custom` - does this mean the outputs of this new version will be different in general?

Reply:
Hi @davidlougheed ,
I am not the original author of the POA code, so I don’t know the reason it was made public.
Both test cases are correct, as the traceback could go either way (same score for both paths). I changed the order of operations and the test case to match the pairwise aligner’s test case, 'test_yclip_suffix_custom()'.
In general, the output should remain the same unless there are two paths with the same score.
Thank you for your feed back and using the new code :).

Reply:
thanks for your response!